testsuite/gnat2goto/tests/type_modular_non_power_of_two_decl/modular_type.adb

```diff
@@ -1,25 +1,19 @@
 procedure Modular_Type is
   type Less_Than_Ten is mod 10;
   procedure Assert_Val (X : Less_Than_Ten) is
-     Y : Less_Than_Ten := X;
-     Z : Less_Than_Ten := Y;
+     Y : Less_Than_Ten := X - 1;
+     Z : Less_Than_Ten := X;
   begin
      pragma Assert (X = 5);
-     Y := X + 3;
-     pragma Assert (Y = 8);
-     Y := Y + 3;
-     pragma Assert (Y = 1);
-     Y := (Y + 1) * X;
-     pragma Assert (Y = 0);
-     Z := Z * 2;
-     pragma Assert (Z = 0);
-     Z := X * 3;
-     pragma Assert (Z = 5);
-     Z := X * Z;
-     pragma Assert (Z = 5);
-     Z := X * 4;
-     pragma Assert (Z = 0);
-     Y := 4;
+
+     pragma Assert (X + 3 = 8);
+     pragma Assert (X + 6 = 1);
+
+     pragma Assert ((X + 7) * X = 0);
+     pragma Assert (X * 2 = 0);
+     pragma Assert (X * 3 = 5);
+     pragma Assert (X * Z = 5);
+     pragma Assert (X * 4 = 0);
```

Comment:
Also another nitpick, would you mind doing these cases in another test file instead of expanding the existing one?

Reply:
Done.